orb/core/connection_types/sql/postgres/statements/select.py

```diff
@@ -37,12 +37,33 @@ def __call__(self, model, context, fields=None):
         sql_columns = defaultdict(list)
         sql_joins = []
 
+        # pre-process columns for expansion based on shortcuts
+        cleaned_columns = []
+        for column in columns:
+            if column.shortcut() and not issubclass(model, orb.View):
+                shortcut = column.shortcut()
+                ref_column = schema.column(shortcut.partition('.')[0])
+                index = None if isinstance(column, orb.ReferenceColumn) else -1
+
+                parts = column.shortcut().split('.')
+                curr = expand
+                for part in parts[:index]:
+                    curr.setdefault(part, {})
+                    curr = curr[part]
+
+                expanded = bool(expand)
+                cleaned_columns.append(ref_column)
+            else:
+                cleaned_columns.append(column)
+
+        columns = cleaned_columns
+
         # process columns to select
         for column in sorted(columns, self.cmpcol):
-            if column.testFlag(column.Flags.Virtual):
+            if column.testFlag(column.Flags.Virtual) and not issubclass(model, orb.View):
```

Comment:
view's were not being properly handled here, since the columns on a database view _are_ virtual, but also defined on the database table itself.